nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/event/pipe/queue/QueueBaseEventBundlePipe.java

```diff
@@ -133,15 +139,24 @@ public void shutdown() throws InterruptedException {
 
     @Override
     public boolean waitForCompletion(long timeoutMillis) throws InterruptedException {
+        final Lock waiting = new ReentrantLock();
+        final Condition flag = waiting.newCondition();
+
         long deadline = System.currentTimeMillis() + timeoutMillis;
         int pause = (int) Math.min(timeoutMillis, 500L);
         do {
             if (queue.size() == 0) {
                 return true;
             }
-            Thread.sleep(pause);
+            waiting.lock();
+            try {
+                flag.await(pause, TimeUnit.MILLISECONDS);
+            } finally {
+                waiting.unlock();
+            }
```

Comment:
Hm no that's not enough, there needs to be cooperation from the people putting/removing things from the queue. Revert to your old code for now (if it works) until this can be fixed, please keep a `// XXX use Condition` for now.